thirdeye-ui/src/app/pages/welcome-onboard-datasource-select-datasource/welcome-onboard-datasource-select-datasource.component.tsx

```diff
@@ -0,0 +1,263 @@
+import type {
+    DatasourceOptionGroups,
+    SelectedDatasource,
+} from "./welcome-onboard-datasource-select-datasource.interfaces";
+import {
+    ADD_NEW_DATASOURCE,
+    getDatasourceGroups,
+} from "./welcome-onboard-datasource-select-datasource.utils";
+
+export const WelcomeSelectDatasource: FunctionComponent = () => {
+    const navigate = useNavigate();
+
+    const [editedDatasource, setEditedDatasource] = useState<Datasource>(
+        createDefaultDatasource()
+    );
+    const [selectedDatasourceName, setSelectedDatasourceName] =
+        useState<SelectedDatasource>(null);
+
+    const { notify } = useNotificationProviderV1();
+    const { t } = useTranslation();
+
+    const handleDatasourceChange = (value: string): void => {
+        setEditedDatasource(JSON.parse(value));
+    };
+
+    const handleRadioChange = (
+        _e: React.ChangeEvent<HTMLInputElement>,
+        value: string
+    ): void => {
+        setSelectedDatasourceName(value as SelectedDatasource);
+    };
+
+    const { datasources, getDatasources, status, errorMessages } =
+        useGetDatasources();
+
+    const datasourceGroups = useMemo<DatasourceOptionGroups[]>(
+        () => getDatasourceGroups(datasources || [], t),
+        [datasources]
+    );
+
+    useEffect(() => {
+        getDatasources();
+    }, []);
+
+    useEffect(() => {
+        notifyIfErrors(
+            status,
+            errorMessages,
+            notify,
+            t("message.error-while-fetching", {
+                entity: t("label.datasources"),
+            })
+        );
+    }, [status]);
+
+    const goToDatasetPage = useCallback(
+        (datasourceName: string) =>
+            navigate(getDataConfigurationCreateDatasetsPath(datasourceName)),
+        []
+    );
+
+    const handleCreateNewDatasource = useCallback(
+        (editedDatasourceProp: Datasource) =>
+            createDatasource(editedDatasourceProp)
+                .then((datasource: Datasource): string => {
+                    notify(
+                        NotificationTypeV1.Success,
+                        t("message.create-success", {
+                            entity: t("label.datasource"),
+                        })
+                    );
+
+                    setSelectedDatasourceName(datasource.name);
+
+                    return datasource.name;
+                })
+                .catch((error: AxiosError): void => {
+                    const errMessages = getErrorMessages(error);
+
+                    isEmpty(errMessages)
+                        ? notify(
+                              NotificationTypeV1.Error,
+                              t("message.create-error", {
+                                  entity: t("label.datasource"),
+                              })
+                          )
+                        : errMessages.map((err) =>
+                              notify(NotificationTypeV1.Error, err)
+                          );
```

Comment:
I recommend using https://github.com/startreedata/thirdeye/blob/d6a261404521d94a37250d4d79fb81d5b0c4284a/thirdeye-ui/src/app/utils/notifications/notifications.util.ts#L19 for these error messages check in this PR. Its used throughout the app

Reply:
Got it. Implemented it here.